src/csharp/ext/grpc_csharp_ext.c

```diff
@@ -197,10 +197,7 @@ void grpcsharp_metadata_array_move(grpc_metadata_array* dest,
 }
 
 GPR_EXPORT void GPR_CALLTYPE
-grpcsharp_batch_context_destroy(grpcsharp_batch_context* ctx) {
-  if (!ctx) {
-    return;
-  }
+grpcsharp_batch_context_reset(grpcsharp_batch_context* ctx) {
```

Comment:
whereas before this was getting called once per `grpcsharp_batch_context` lifecycle, IIC this is now getting called twice: once when it is first recycled to the pool, and then a second time when it's pool later disposes it. Tests are fine, but it seems unexpected that that's ok?

Reply:
Yes, reset is indeed getting called twice. This is to make the scenarios where you aren't pooling the context objects still work.
- if you aren't pooling you see no overhead - you need to reset anyway to destroy the child objects
- if you are pooling, the you're hopefully reusing the object many times, so the overhead of calling reset twice in the last round should be negligible.